synapse/rest/client/v2_alpha/account.py

```diff
@@ -499,13 +501,23 @@ class EmailThreepidRequestTokenRestServlet(RestServlet):
     PATTERNS = client_patterns("/account/3pid/email/requestToken$")
 
     def __init__(self, hs):
-        self.hs = hs
         super(EmailThreepidRequestTokenRestServlet, self).__init__()
-        self.identity_handler = hs.get_handlers().identity_handler
+        self.hs = hs
+        self.config = hs.config
         self.store = self.hs.get_datastore()
+        self.identity_handler = hs.get_handlers().identity_handler
 
     @defer.inlineCallbacks
     def on_POST(self, request):
+        if self.config.threepid_behaviour == "off":
+            if self.config.local_threepid_emails_disabled_due_to_config:
+                logger.warn(
+                    "User password resets have been disabled due to lack of email config"
```

Comment:
This seems to be in conflict with the error at line 518... is this servlet used for password resets, adding identities, or both?

Reply:
Actually, this whole section shouldn't even have these checks as this PR should only deal with registration.

Reply:
even better :)